File: Valid Parentheses/solution_replace.py
def is_valid_replace(s: str) -> bool:
    """Return True if the bracket string is valid using iterative replacement."""

    previous = None
    current = s

    while previous != current:
        previous = current
        current = current.replace("()", "").replace("{}", "").replace("[]", "")

    return current == ""

File: Valid Parentheses/solution_replace.js
/**
 * Iterative replacement strategy for validating parentheses strings.
 * @param {string} s
 * @returns {boolean}
 */
function isValidReplace(s) {
  let prev = null;
  let current = s;

  while (prev !== current) {
    prev = current;
    current = current.replace(/\(\)/g, "").replace(/\{\}/g, "").replace(/\[\]/g, "");
  }

  return current.length === 0;
}

module.exports = { isValidReplace };
